borsh.go

```diff
@@ -379,33 +379,33 @@ func serialize(v reflect.Value, b io.Writer) error {
 	var err error
 	switch v.Kind() {
 	case reflect.Int8:
-		_, err = b.Write([]byte{byte((v.Interface().(int8)))})
+		_, err = b.Write([]byte{byte((v.Int()))})
 	case reflect.Int16:
 		tmp := make([]byte, 2)
-		binary.LittleEndian.PutUint16(tmp, uint16(v.Interface().(int16)))
+		binary.LittleEndian.PutUint16(tmp, uint16(v.Int()))
 		_, err = b.Write(tmp)
 	case reflect.Int32:
 		tmp := make([]byte, 4)
-		binary.LittleEndian.PutUint32(tmp, uint32(v.Interface().(int32)))
+		binary.LittleEndian.PutUint32(tmp, uint32(v.Int()))
 		_, err = b.Write(tmp)
 	case reflect.Int64:
 		tmp := make([]byte, 8)
-		binary.LittleEndian.PutUint64(tmp, uint64(v.Interface().(int64)))
+		binary.LittleEndian.PutUint64(tmp, uint64(v.Int()))
 		_, err = b.Write(tmp)
 	case reflect.Int:
 		tmp := make([]byte, 8)
-		binary.LittleEndian.PutUint64(tmp, uint64(v.Interface().(int)))
+		binary.LittleEndian.PutUint64(tmp, uint64(v.Int()))
 		_, err = b.Write(tmp)
```

Comment:
This can cause overflow on 32-bit platforms. The size of `int` is platform-dependent, and it's unsafe to cast `int64` value into `int` or `uint`.

Actually `int` and `uint` are just unsafe to use in `borsh-go` structs. Perhaps we should just get rid of them.

Reply:
yes, you are right. Thank for your reviewing!
I have reverted the modification of reflect.Int.